File: easy/ShortestDistanceToCharacter.js
// Given a string s and a character c that occurs in s, return an array of integers answer where answer.length == s.length and answer[i] is the distance from index i to the closest occurrence of character c in s.

// The distance between two indices i and j is abs(i - j), where abs is the absolute value function.

const shortestToChar = function(s, c) {
  const arrOfC = [];
  const resultArr = [];
  for (let i = 0; i < s.length; i++) {
    if (s[i] === c) {
      arrOfC.push(i)
    }
  }
  for (let i = 0; i < s.length; i++) {
    let smallestDist = Infinity;
    for (let j = 0; j < arrOfC.length; j++) {
      let dist = Math.abs(arrOfC[j] - i)
      if (dist === 0) {
        smallestDist = dist;
        break;
      }
      if (dist < smallestDist) {
        smallestDist = dist;
      }
    }
    resultArr.push(smallestDist)
  }
  return resultArr;
};

const testS = "loveleetcode";
const testC = "e";

console.log(shortestToChar(testS, testC))

File: easy/ShortestDistanceToCharacter.py
# Given a string s and a character c that occurs in s, return an array of integers answer where answer.length == s.length and answer[i] is the distance from index i to the closest occurrence of character c in s.

# The distance between two indices i and j is abs(i - j), where abs is the absolute value function.

def shortestToChar(s, c) :
  (arr, location) = ([], [])
  i = 0
  for letter in s :
    if letter == c :
      location.append(i)
    i += 1
  j = 0
  while j < len(s) :
    shortest = float('inf')
    for num in location :
      diff = abs(num - j)
      if diff < shortest :
        shortest = diff
        if shortest == 0 :
          break
    arr.append(shortest)
    j += 1
  return arr


testS = "loveleetcode"
testC = "e"

print(shortestToChar(testS, testC))